metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/MediaExtractorImpl.java

```diff
@@ -102,6 +108,18 @@ public MediaExtractorImpl(int redirectCount, int thumbnailGenerateTimeout,
     this.textProcessor = new TextProcessor(thumbnailGenerator,
         new PdfToImageConverter(new CommandExecutor(thumbnailGenerateTimeout)));
     this.media3dProcessor = new Media3dProcessor();
+    this.oEmbedProcessor = new OEmbedProcessor();
+    this.linkedProcessor = new LinkedProcessor();
+    initChainOfProcessors();
+  }
+
+  private void initChainOfProcessors() {
```

Comment:
The reason I chose this pattern is to create the structure to be able to chain processors. In this case creates the possibility linked->oembed->text. Or in the future we want to have specific image processors they can be chained. image->2dimage->3dimage or audio processors audioA->audioB->audioC. if one processor cannot handle it passes to the next one, and so on.

Reply:
yes. you are right at this point only the structure is introduced. the chaining feature needs to be coded. Another reason I opted to do it this way was to reuse the existing code. otherwise I will duplicate code. From my point of view I don't see it complex yet. If it get like this I do myself responsible to maintain it and later we can refactor it if there is no needed anymore. what do you think?

Reply:
It is not a complexity that cannot be handled but it invites unnecessary complexity even if it is low now.
Honestly if a chain of processors is needed I would not do it this way since each processor needs to be aware of the "next" processor(even if it is not needed). I would create a structure that would have the chaining in a list that would be iterated afterwards(the processors themselves are not aware of the chaining).
But given the current use case it feels to me at least that it is a simple `ifelse` that could be integrated in the text processor.

In fact the first time I read the code I thought the oEmbedProcessor should create some content that should be available to the TextProcessor that should run afterwards but then I realized it is this or that case.

What do you think?

P.S: I appreciate you offering to be responsible of the code, but what if you for example are sick or leave the organization? xD :D

Reply:
I like the Idea of removing the chaining from the processors, but this means the chaining goes to an upper level class to orchestrate the processors, I don't know yet if that will be possible because how media processing is implemented, need to think another way.

Actually the list is created on the initChainProcessors method. That is how they become aware of each other.

The thing is that I don't wanted to mix the TextProcessor with the oEmbedProcessor. It will bloat the class TextProcessor doing two different things at a time breaking the single responsability principle.

PS: That is why I created the LinkedProcessor lol. because I don't want to mix the processors, I want them to be pluggables. the challenge with the processors is that they don't respond to mimetype, but to this generic structure of AUDIO, IMAGE, 3D, TEXT, OTHER the way media processing is coded narrow me to go this  way.